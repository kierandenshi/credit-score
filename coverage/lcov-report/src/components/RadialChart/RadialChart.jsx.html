<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RadialChart/RadialChart.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RadialChart</a> RadialChart.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/14</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './_RadialChart.scss';
&nbsp;
class RadialChart extends Component {
    componentDidMount() {
        const { value, maxValue } = this.props;
        const r = (180 / 100) * ((value / maxValue) * 100);
        setTimeout(<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >            this.fullMask.style.transform = `rotate(${r}deg)`;</span>
<span class="cstat-no" title="statement not covered" >            this.fullMaskFill.style.transform = `rotate(${r}deg)`;</span>
<span class="cstat-no" title="statement not covered" >            this.halfMaskFill.style.transform = `rotate(${r}deg)`;</span>
<span class="cstat-no" title="statement not covered" >            this.halfMaskFillFix.style.transform = `rotate(${r * 2}deg)`;</span>
        }, 1000);
    }
&nbsp;
    render() {
        return (
            &lt;div className={'radial-chart'}&gt;
                &lt;div className={'circle'}&gt;
                    &lt;div className={'mask full'} ref={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; { <span class="cstat-no" title="statement not covered" >this.fullMask = e; </span>}}&gt;
                        &lt;div className={'fill'} ref={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; { <span class="cstat-no" title="statement not covered" >this.fullMaskFill = e; </span>}} /&gt;
                    &lt;/div&gt;
                    &lt;div className={'mask half'}&gt;
                        &lt;div className={'fill'} ref={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; { <span class="cstat-no" title="statement not covered" >this.halfMaskFill = e; </span>}} /&gt;
                        &lt;div className={'fill fix'} ref={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; { <span class="cstat-no" title="statement not covered" >this.halfMaskFillFix = e; </span>}} /&gt;
                    &lt;/div&gt;
                &lt;/div&gt;
                &lt;div className={'inset'}&gt;
                    {this.props.children}
                &lt;/div&gt;
            &lt;/div&gt;
        );
    }
}
&nbsp;
RadialChart.defaultProps = {
    children: null,
};
&nbsp;
RadialChart.propTypes = {
    maxValue: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
    children: PropTypes.node,
};
&nbsp;
export default RadialChart;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 16 2018 09:27:06 GMT+0000 (GMT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
