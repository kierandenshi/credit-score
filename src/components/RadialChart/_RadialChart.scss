@import '~@css/includes/helpers';

.radial-chart {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;

  svg {
    circle {
      stroke: $font-color;
    }
  }
}
