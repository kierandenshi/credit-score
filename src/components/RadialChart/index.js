export { default } from './RadialChart.jsx';
