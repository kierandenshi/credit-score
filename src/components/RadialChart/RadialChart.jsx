import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import './_RadialChart.scss';

const RadialChart = ({
    maxValue,
    value,
    children,
}) => {
    // Radius = (size / 2) - (stroke / 2)
    // Circumference = 2 * Ï€ * Radius
    // The length of the offset is inversely proportional to the amount of progress.
    // So to display 60 % progress, the offset as 40 % of the circumference.
    const perc = value / maxValue;

    const size = 320;
    const stroke = 8;
    const radius = (size / 2) - (stroke / 2);
    const dashArray = (2 * Math.PI * radius);
    const dashOffset = (2 * Math.PI * radius) * (1 - perc);


    const fill = keyframes`
        0% {
            stroke-dashoffset: ${dashArray};
        }
        100% {
            stroke-dashoffset: ${dashOffset};
        }
    `;

    const AnimatedChart = styled.svg`
        transform: rotate(-90deg);
        & circle {
            stroke-dasharray: ${dashArray};
            animation: ${fill} 2s forwards;
        }
    `;

    return (
        <div className={'radial-chart'}>
            <AnimatedChart width={size} height={size} viewBox={`0 0 ${size} ${size}`}>
                <circle
                    cx={size / 2}
                    cy={size / 2}
                    r={radius}
                    fill={'none'}
                    strokeWidth={stroke}
                />
            </AnimatedChart>
        </div>
    );
};

RadialChart.defaultProps = {
    children: null,
};

RadialChart.propTypes = {
    maxValue: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
    children: PropTypes.node,
};

export default RadialChart;
