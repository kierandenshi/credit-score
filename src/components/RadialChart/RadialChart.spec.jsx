import React from 'react';
import { shallow } from 'enzyme';
import RadialChart from './RadialChart.jsx';

describe('RadialChart component', () => {
    const wrapper = shallow(<RadialChart />);
    it('should render', () => {
        expect(wrapper.find('.radial-chart')).toMatchSnapshot();
    });
});
