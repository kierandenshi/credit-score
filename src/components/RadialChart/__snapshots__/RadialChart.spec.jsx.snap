// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadialChart component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <RadialChart />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.svg
          height={320}
          viewBox="0 0 320 320"
          width={320}
        >
          <circle
            cx={160}
            cy={160}
            fill="none"
            r={156}
            strokeWidth={8}
          />
        </styled.svg>,
        "className": "radial-chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <circle
            cx={160}
            cy={160}
            fill="none"
            r={156}
            strokeWidth={8}
          />,
          "height": 320,
          "viewBox": "0 0 320 320",
          "width": 320,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cx": 160,
            "cy": 160,
            "fill": "none",
            "r": 156,
            "strokeWidth": 8,
          },
          "ref": null,
          "rendered": null,
          "type": "circle",
        },
        "type": [Function],
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <styled.svg
            height={320}
            viewBox="0 0 320 320"
            width={320}
          >
            <circle
              cx={160}
              cy={160}
              fill="none"
              r={156}
              strokeWidth={8}
            />
          </styled.svg>,
          "className": "radial-chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <circle
              cx={160}
              cy={160}
              fill="none"
              r={156}
              strokeWidth={8}
            />,
            "height": 320,
            "viewBox": "0 0 320 320",
            "width": 320,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": 160,
              "cy": 160,
              "fill": "none",
              "r": 156,
              "strokeWidth": 8,
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.svg
        height={320}
        viewBox="0 0 320 320"
        width={320}
      >
        <circle
          cx={160}
          cy={160}
          fill="none"
          r={156}
          strokeWidth={8}
        />
      </styled.svg>,
      "className": "radial-chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <circle
          cx={160}
          cy={160}
          fill="none"
          r={156}
          strokeWidth={8}
        />,
        "height": 320,
        "viewBox": "0 0 320 320",
        "width": 320,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cx": 160,
          "cy": 160,
          "fill": "none",
          "r": 156,
          "strokeWidth": 8,
        },
        "ref": null,
        "rendered": null,
        "type": "circle",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.svg
          height={320}
          viewBox="0 0 320 320"
          width={320}
        >
          <circle
            cx={160}
            cy={160}
            fill="none"
            r={156}
            strokeWidth={8}
          />
        </styled.svg>,
        "className": "radial-chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <circle
            cx={160}
            cy={160}
            fill="none"
            r={156}
            strokeWidth={8}
          />,
          "height": 320,
          "viewBox": "0 0 320 320",
          "width": 320,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cx": 160,
            "cy": 160,
            "fill": "none",
            "r": 156,
            "strokeWidth": 8,
          },
          "ref": null,
          "rendered": null,
          "type": "circle",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
